### PR TITLE
Rails Blog App: (6) Views

### PR DESCRIPTION
**Pull request by adarachel**
### In this Milestone:

Project requirements
Implement the design from the sneak peek wireframes.
Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,25 +1,14 @@
...
-    author.update(posts_counter: author.posts.count)
-  end
-
-  def five_most_recent_comments
-    comments.order(created_at: :desc).limit(5)
-  end
+  validates :title, presence: true
+  validates :author, presence: true
+  validates :title, length: { in: 1..250 }
+  validates :likes_counter, :comment_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] [OPTIONAL]
The error below is thrown in an attempt to add post for a current user. We resolve that by adding `allow_blank: true` to the validations of `likes_counter` and `comments_counter`

**Error adding user post data:**
image

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<h1><%= @user.name %></h1>
-<p><%= @user.bio %></p>
-<!-- Display user's details -->
\ No newline at end of file
+<div class="user_container">
+  <%= render 'shared/user_profile', user: @user %>
+</div>
```
Comment (rloterh) [inline app/views/users/show.html.erb:3]: - [ ] All posts of a user are appearing on the user's page, however we want to show only the 3 most recent post. We can achieve that by defining 3 most  `recent_post` method in the users model to render the user's 3 most recent post in the user show view, like the example below:

```ruby
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

  <h4><%= link_to post.title, user_post_path(post) %></h4>
  <p><%= post.text %></p>
  <p>
    <small>Comments: <%= post.comments_counter || 0 %>,</small> 
    <small>Likes: <%= post.likes_counter || 0 %></small>
  </p>

<% end %>

<button>
  <%= link_to 'See all Post', user_posts_path(@user) %>
</button>
```

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+    <a href="/users/<%= post.author_id%>/posts/<%= post.id %>"><%= post.title%></a>
+        <p>
+           <%= post.text%>
+        </p>
+        <p class="comments_likes">
+            Comments: <%= post.comment_counter %> Likes: <%= post.likes_counter%>
+        </p>
+</div>
+<div class="boxed">
+<% Post.get_5_comments(post).each  do |comment|%>
```
Comment (rloterh) [inline app/views/posts/index.html.erb]: - [ ] To ensure that only 5 recent comments are displayed:

image

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
-<ul>
-  <% @users.each do |user| %>
-    <li><%= user.name %></li>
-  <% end %>
-</ul>
+
+<% @users.each do |user| %>
+
+  <%= render 'shared/user_profile', user: user %>
+
```
Comment (rloterh) [inline app/views/users/index.html.erb:5]: Let's ensure our views conform to the wireframe design found on the [project instructions page](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md).

image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rachel, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁  🌴 🚦 🚗 🏫  🏡   _🏃‍♀️

## Highlights 👍  
- Good job so far on views creations!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (peterrobert) [reply app/models/post.rb]: This change has been done well

**Review by victorokoroji — APPROVED** [review]
Hi @adarachel , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**